src/luascript.cpp

```diff
@@ -564,17 +565,20 @@ void LuaScriptInterface::callVoidFunction(int params)
 void LuaScriptInterface::pushVariant(lua_State* L, const LuaVariant& var)
 {
 	lua_createtable(L, 0, 2);
-	setField(L, "type", var.type);
-	switch (var.type) {
+	setField(L, "type", var.index());
+	switch (var.index()) {
 		case VARIANT_NUMBER:
-			setField(L, "number", var.number);
+			setField(L, "number", std::get<VARIANT_NUMBER>(var));
 			break;
 		case VARIANT_STRING:
-			setField(L, "string", var.text);
+			setField(L, "string", std::get<VARIANT_STRING>(var));
 			break;
 		case VARIANT_TARGETPOSITION:
+			pushPosition(L, std::get<VARIANT_TARGETPOSITION>(var));
+			lua_setfield(L, -2, "pos");
+			break;
 		case VARIANT_POSITION: {
-			pushPosition(L, var.pos);
+			pushPosition(L, std::get<VARIANT_POSITION>(var));
 			lua_setfield(L, -2, "pos");
 			break;
 		}
```

Comment:
It's also possible to [`visit`](https://en.cppreference.com/w/cpp/utility/variant/visit) the variant instead. I went with the approach that changed as little as possible of the code.